blocks/library/embed/index.js

```diff
@@ -26,7 +27,25 @@ import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 // These embeds do not work in sandboxes
 const HOSTS_NO_PREVIEWS = [ 'facebook.com' ];
 
-function getEmbedBlockSettings( { title, description, icon, category = 'embed', transforms, keywords = [] } ) {
+// Caches the embed API calls, so if blocks get transformed, or deleted and added again, we don't spam the API.
+const wpEmbedAPI = memoize( ( url ) => wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ) );
+
+const matchesPatterns = ( url, patterns ) => {
+	return patterns.some( ( pattern ) => {
+		return url.match( pattern );
+	} );
+};
+
+const findBlock = ( url ) => {
+	for ( const block of [ ...common, ...others ] ) {
+		if ( matchesPatterns( url, block.patterns ) ) {
+			return block.name;
+		}
+	}
+	return 'core/embed';
+};
+
+function getEmbedBlockSettings( { title, description, icon, category = 'embed', transforms, keywords = [], patterns = [] } ) {
```

Comment:
We're no longer passing `patterns` here, so all of the logic related to this, specifically on line 130, will not be triggered. In practice, it's not an issue because `findBlock` will get triggered and find the correct block anyways, but seems we might not need this as a parameter.

Reply:
ah yes, removing it now!